src/lib/networks/Network.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  computed,
+  reactive,
+  ref,
+} from '@vue/composition-api';
+import type {
+  IAccount,
+  INetworkFeatures,
+  INetworkAdapter,
+  INetworkInstance,
+  INetworkMethods,
+  INetworkState,
+  SupportedNetworkType,
+  INetworkSettings,
+} from '../../types';
+import { AeternityNetwork } from './aeternity/AeternityAdapter';
+import { NETWORK_MAINNET, NETWORK_TESTNET } from '../../popup/utils';
+import { useModals } from '../../composables';
+
+export const Network = (() => {
+  const { openDefaultModal } = useModals();
+
+  const networkAdapters: Record<SupportedNetworkType, INetworkAdapter> = {
+    aeternity: AeternityNetwork,
+    bitcoin: AeternityNetwork, // TODO for testing purposes
+  };
+
+  const networkSettingsList = ref<INetworkSettings[]>([
+    { ...NETWORK_MAINNET, type: 'aeternity' },
+    { ...NETWORK_TESTNET, type: 'aeternity' },
+    { ...NETWORK_MAINNET, type: 'bitcoin', name: 'BTC Test' }, // TODO for testing purposes
+  ]);
+
+  function getEmptyNetworkState(): INetworkState {
+    return {
+      isReady: ref(false),
+      isConnected: ref(false),
+      isError: ref(false),
+      accounts: ref([]),
+      activeAccountIdx: ref(0),
+      activeAccount: computed(() => ({} as IAccount)),
```

Comment:
Why network is storing the information about accounts?

Reply:
Good question :) I know that this might look like the `Network` lib is holding this states, but it's not fully true.

It is only passing the states received by the adapter to the places that requires the access. So before having any network running we have a moment where the components should have access to the state (`Network.current.state`), the state should have the reactive props in place, and the values should be "empty".

`getEmptyNetworkState` creates this empty state, which is filled in (replaced) by the `setNetworkState` at the moment of network adapter initialization.

Is this clear? If not I think we could try to update this/